### PR TITLE
chore: Port MPIHasher to Swift

### PR DESCRIPTION
**Pull request by einsteinx2**
## Summary
 - Now that MPHasher C++ code has been refactored out and the MPIHasher Obj-C class updated, it was a prime candidate for a quick port to Swift as part of our ongoing porting efforts. This builds off of my previous PR here: https://github.com/mParticle/mparticle-apple-sdk/pull/233

 ## Testing Plan
 - Confirmed that the hash functions produce the same output using our comprehensive unit tests
 - Smoke tested with a local app

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5908

### CODE REVIEW

**Comment by einsteinx2** [comment]
After the other PR is merged, I will rebase this branch off of development, change the target branch for the PR to development, and switch to a real PR from a draft.

File: UnitTests/HasherTests.m [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+#else
+@import mParticle_Apple_SDK_NoLocation;
+#endif
+
 #import <XCTest/XCTest.h>
 #import "MPEnums.h"
 #import "MPEvent.h"
 #import "MPBaseTestCase.h"
-#import "MPIHasher.h"
 #import "mparticle.h"
```
Comment (BrandonStalnaker) [inline UnitTests/HasherTests.m:11]: if you're changing to module import couldn't you also remove MPEvent.h, mparticle.h, and MPEnums.h?

**Review by BrandonStalnaker — APPROVED** [review]
Just a nitpick of a comment. Otherwise LGTM

File: UnitTests/HasherTests.m (hunk 1 above)
Comment (einsteinx2) [reply UnitTests/HasherTests.m:11]: Good point, any public headers should not need to be imported anymore, though this is just a unit test so doesn't really matter. I'll see if I can clean it up real quick.

Comment (einsteinx2) [reply UnitTests/HasherTests.m:11]: actually I'm going to close this and re-open a new one for the real PR. I'll include this change.

**Pull request closed without merging**